org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/desktop/datachange/ActiveFormDataChangeManager.java

```diff
@@ -0,0 +1,62 @@
+package org.eclipse.scout.rt.client.ui.desktop.datachange;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.scout.rt.client.ui.desktop.DesktopEvent;
+import org.eclipse.scout.rt.client.ui.desktop.IDesktop;
+import org.eclipse.scout.rt.client.ui.form.FormEvent;
+import org.eclipse.scout.rt.client.ui.form.IForm;
+import org.eclipse.scout.rt.platform.ApplicationScoped;
+import org.eclipse.scout.rt.platform.BEANS;
+
+/**
+ * Data change manager which may be used to avoid data-change notifications for hidden forms.
+ * Events for listeners registered to this manager will be buffered until the form is activated.
+ */
+@ApplicationScoped
+public class ActiveFormDataChangeManager implements IDataChangeListener {
+
+  protected Map<IForm, IDataChangeManager> m_dataChangeManagers = new HashMap<>();
+
+  public ActiveFormDataChangeManager() {
+    IDesktop desktop = IDesktop.CURRENT.get();
```

Comment:
ApplicationScoped beans are singletons, the constructor is called only once for every session -> This will work only the first time.
You could move the dataChangeManagers to the clientSession. Then add a getter on this bean getDataChangeManagers () and create the map and set it using clientSession.setData the first time the getter is called for a session.